core/src/main/java/org/elasticsearch/snapshots/SnapshotShardFailure.java

```diff
@@ -215,6 +216,11 @@ public static SnapshotShardFailure fromXContent(XContentParser parser) throws IO
             throw new ElasticsearchParseException("index shard was not set");
         }
         snapshotShardFailure.shardId = new ShardId(index, index_uuid, shardId);
+        // Workaround for https://github.com/elastic/elasticsearch/issues/25878
+        // Some old snapshot might still have null in shard failure reasons
+        if (snapshotShardFailure.reason == null) {
+            snapshotShardFailure.reason = "";
```

Comment:
What I don't quite understand: Why will it happily parse the `reason` field if it is null? Currently we parse it using `text()`, shouldn't that fail and should we use `textOrNull()` instead?